### PR TITLE
Serve correct mime type on release dev server

### LINKED ISSUE #43904: Flutter Web - Unable to load css assets from html in release mode
I have a simple HTML file in which I load my CSS as a <link> tag in the head. Everything works fine in debug mode using the command `flutter run -d chrome` (debug) but not using `flutter run -d chrome --release`. In this situation, I get the following error:

```
Refused to apply style from 'styles.css' because its MIME type ('text') is not a supported stylesheet MIME type, and strict MIME checking is enabled.
```

After doing some investigation (see [this thread](https://stackoverflow.com/questions/48248832/stylesheet-not-loaded-because-of-mime-type) on Stack Overflow), I see the problem seems to come from the fact that the response header for the css file returns a content-type "text" instead of "text/css". If I specify `type="text"` in my link tag instead, I get rid of the error but the css isn't loaded either.

### HTML Source
``` html
<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8" />
        <title>MyApp</title>
        <script src="main.dart.js"></script>
	<link rel="stylesheet" type="text/css" href="styles.css" />
    </head>
    <body></body>
</html>
```

Before you ask, the file is present, it's not a loading error.

## Steps to Reproduce
Just start the app with the following command: `flutter run -d chrome --release`

**Target OS version/browser:** Chrome

### flutter analyze
```
Analyzing live_app...

   info - Unused import: 'package:live_app/util/logger/logger.dart' - lib\app\vroom\vroom_app.dart:9:8 - unused_import
   info - Unused import: 'package:live_app/util/logger/logger.dart' - lib\core\base_core.dart:16:8 - unused_import
   info - Unused import: 'package:live_app/util/logger/logger.dart' - lib\managers\login\server_info_step.dart:6:8 - unused_import
   info - The value of the field '_iceServers' isn't used - lib\managers\stream\webrtc\base_webrtc_peer.dart:30:17 - unused_field
   info - The value of the field '_mediaConstraints' isn't used - lib\managers\stream\webrtc\base_webrtc_peer.dart:58:24 - unused_field
   info - The declaration '_onGetStreamSuccess' isn't referenced - lib\managers\stream\webrtc\base_webrtc_peer.dart:117:8 - unused_element
   info - The declaration '_getDisplayMedia' isn't referenced - lib\managers\stream\webrtc\base_webrtc_peer.dart:273:8 - unused_element
   info - Unused import: 'package:meta/meta.dart' - lib\managers\user\user_manager.dart:1:8 - unused_import
   info - The declaration '_showAudioDialog' isn't referenced - lib\screens\vroom\display\screen_sharing_display.dart:153:8 - unused_element
   info - The value of the local variable 'core' isn't used - lib\screens\vroom\display\screen_sharing_display.dart:161:17 - unused_local_variable
   info - This function has a return type of 'Map<String, dynamic>', but doesn't end with a return statement - lib\util\webrtc\webrtc.dart:45:10 - missing_return
   info - This function has a return type of 'Map<String, dynamic>', but doesn't end with a return statement - lib\util\webrtc\webrtc.dart:46:10 - missing_return
   info - This function has a return type of 'Future<List>', but doesn't end with a return statement - lib\util\webrtc\webrtc.dart:49:10 - missing_return
   info - This function has a return type of 'Future<List>', but doesn't end with a return statement - lib\util\webrtc\webrtc.dart:50:10 - missing_return
   info - This function has a return type of 'Future<List>', but doesn't end with a return statement - lib\util\webrtc\webrtc.dart:51:10 - missing_return
   info - `Future` results in `async` function bodies must be `await`ed or marked `unawaited` using `package:pedantic` - test\main_test.dart:19:7 - unawaited_futures
   info - `Future` results in `async` function bodies must be `await`ed or marked `unawaited` using `package:pedantic` - test_driver\app_test.dart:21:9 - unawaited_futures

17 issues found. (ran in 3.1s)
```

### flutter doctor -v
```
[√] Flutter (Channel master, v1.10.15-pre.272, on Microsoft Windows [Version 10.0.17763.805], locale fr-CA)
    • Flutter version 1.10.15-pre.272 at D:\_utils\Flutter
    • Framework revision 5bc7fb6835 (3 days ago), 2019-10-27 21:25:13 -0700
    • Engine revision 3ebf006904
    • Dart version 2.6.0 (build 2.6.0-dev.8.2 6354b0b97d)


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
    • Android SDK at C:\Users\nicolas.dion\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.0
    • ANDROID_HOME = C:\Users\nicolas.dion\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[!] Android Studio (version 3.4)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[!] IntelliJ IDEA Ultimate Edition (version 2016.3)
    • IntelliJ at C:\Program Files (x86)\JetBrains\IntelliJ IDEA 2016.3.8
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins
    X This install is older than the minimum recommended version of 2017.1.0.

[!] IntelliJ IDEA Ultimate Edition (version 2019.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2019.2
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[√] VS Code, 64-bit edition (version 1.39.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.5.1

[√] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 78.0.3904.70
    • Web Server • web-server • web-javascript • Flutter Tools

! Doctor found issues in 3 categories.
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fixes https://github.com/flutter/flutter/issues/43904

### CODE REVIEW

**Review by zanderso — APPROVED** [review]
lgtm on green

File: packages/flutter_tools/test/general.shard/web/asset_server_test.dart [hunk 1]
```diff
@@ -78,4 +78,18 @@ void main() {
 
     expect(response.statusCode, 404);
   }));
+
+  test('release asset server serves correct mime type and content length', () => testbed.run(() async {
```
Comment (mdebbar) [inline packages/flutter_tools/test/general.shard/web/asset_server_test.dart]: Few comments about the tests:
- Does it make sense to have a test that exercises the inference based on `headerBytes`? (i.e. omit the file extension).
- Can we add tests for the most important mime types (`text/javascript` and `text/html`)?

**Review by mdebbar — APPROVED** [review]
LGTM!

File: packages/flutter_tools/test/general.shard/web/asset_server_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/web/asset_server_test.dart]: Updated to add more test cases

Comment (mdebbar) [reply packages/flutter_tools/test/general.shard/web/asset_server_test.dart]: Thanks!

**Review by harryterkelsen — APPROVED** [review]

**Pull request merged**